rewrite-pattern.js

```diff
@@ -256,6 +256,10 @@ const processTerm = (item, regenerateOptions, groups) => {
 			break;
 		case 'value':
 			const codePoint = item.codePoint;
+			if (item.kind === "symbol" && codePoint >= 0x20 && codePoint <= 0x7E) {
```

Comment:
Let's keep this as `codePoint == 0x2D`, since many ascii symbols must be escaped: `[`, `{', `|`, `+`, `?`, `*`, `$`, ...

Reply:
Escaped character, i.e. `\[` is `identifier` kind instead of `symbol` kind. If `regjsparser` parses it safely as `symbol`, we don't have to escape it.